internal/helm/helm.go

```diff
@@ -390,8 +390,9 @@ func Status(ctx context.Context, logger logr.Logger, cluster *kubernetes.Cluster
 var syncNamespaceClientMap sync.Map
 
 type SynchronizedClient struct {
-	namespace  string
-	m          sync.Mutex
+	namespace string
+	// mutexMap is holding the mutexes for the same releases
+	mutexMap   sync.Map
```

Comment:
Yes, unfortunately it doesn't seems to be easy to understand when to cleanup some resources, because you cannot cleanup the map at the end of a release:

Example with a concurrent release (3 times) with a duration of 5s (total 15s)
```
t0) start of release, mutex-A not present, loaded/stored and locked
-
t3) try to start new release, mutex-A loaded from map, try to acquire lock, wait
-
t5) first release end, mutex-A unlocked and deleted from map, locked acquired from same mutex-A, start release
// note: mutex-A is not in map anymore, first release deleted after unlock

t7) try to start a new release, loading mutex-B from map because key was empty/deleted, start concurrent release

: (
```

Not sure if there is an easy way to handle this.. a mutex should not be a big issue but yes, I can see the problem.

I would say to leave it like this, and see if we can track, cleanup this somehow.
We can probably at least safely the lock after an `UninstallRelease`. What do you think?

Reply:
Added cleanup on uninstall. :slightly_smiling_face:

Reply:
Removed cleanup!